### PR TITLE
Suggestion for https://github.com/huggingface/optimum-quanto/pull/263

### PR DESCRIPTION
**Pull request by Wauplin**
This PR is there to showcase what I suggested in https://github.com/huggingface/optimum-quanto/pull/263#pullrequestreview-2208084454 cc @sayakpaul 

In particular, it leverages the existing `ModelHubMixin` from `huggingface_hub`:
- it avoids to define a `PushToHubMixin` since the `push_to_hub` method is built-in `ModelHubMixin` 
- `_save_pretrained` has to be defined to serialize the model. The `save_pretrained` method calls it + generates a basic model card
- `from_pretrained` has to be defined since we cannot really leverage `ModelHubMixin.from_pretrained` in this context (too many difference).

Regarding the model card, there are two ways to made it more complete:
- define high level metadata like library_name, tags, repo_url, etc. in the class definition. This is straightforward and will help having all `optimum-quanto` models tagged as such on the Hub (and therefore searchable). => not done in this PR but low hanging fruit and highly recommended
- and / or redefine `def generate_model_card(self, *args, **kwargs) -> ModelCard:` in the class inheritance => allows more things but maybe for a follow-up PR if needed

Related documentations:
- for `ModelHubMixin`: https://huggingface.co/docs/huggingface_hub/guides/integrations#a-more-complex-approach-class-inheritance
- for model cards specifically: https://huggingface.co/docs/huggingface_hub/guides/integrations#model-card

---

A side effect of this PR is that `max_shard_size` parameter in `save_pretrained` is lost. I don't think it's a problem to have it hard-coded to set a good standard on the Hub. Parameters that are rarely used are usually more costly than beneficial.  

Disclaimer: I haven't tested the PR locally. There might be a few things to tweak but the main idea is in there.

### CODE REVIEW

File: optimum/quanto/models/diffusers_models.py [hunk 1]
```diff
@@ -177,8 +176,8 @@ def from_pretrained(cls, model_name_or_path: Union[str, os.PathLike], **kwargs):
         model.eval()
         return cls(model)
 
-    def save_pretrained(self, save_directory: Union[str, os.PathLike], max_shard_size: Union[int, str] = "10GB"):
-        self._wrapped.save_pretrained(save_directory, max_shard_size=max_shard_size)
+    def _save_pretrained(self, save_directory: Path) -> None:
```
Comment (sayakpaul) [inline optimum/quanto/models/diffusers_models.py:179]: Is this called internally by `ModelMixin`? 

You have already mentioned that we're losing access to `max_shard_size` but why does it have to be that?

**Review by sayakpaul — APPROVED** [review]
Thanks for the changes! Pretty solid. 

I think disabling `max_shard_size` should be okay as `quanto` users haven't yet used this argument.

**Comment by sayakpaul** [comment]
@Wauplin, we need to squash the commits so that we can make conversational commits happy.

File: optimum/quanto/models/diffusers_models.py (hunk 1 above)
Comment (Wauplin) [reply optimum/quanto/models/diffusers_models.py:179]: Yes it is. This guide explains you how that works: https://huggingface.co/docs/huggingface_hub/guides/integrations#a-more-complex-approach-class-inheritance. Basically by using the mixin you benefit from a few built-in stuff but on the other hand you have to comply to some constraints. You cannot add arguments here for instance. I'd say it's better like this to have more uniformization in signatures in the ecosystem.

Comment (sayakpaul) [reply optimum/quanto/models/diffusers_models.py:179]: Not a strong opinion but libraries may have constraints beyond enforcements. But I understand how that kind of uniformization could be beneficial.

**Comment by Wauplin** [comment]
@sayakpaul I force-pushed the squash commit

**Comment by Wauplin** [comment]
I had to create a fork to open this PR so definitely no privileges here :smile: I let you handle it from this point :wink:

**Pull request merged**